### PR TITLE
fix: Eigen Identity matrix and Vc/SMatrix matrix getter and hide arithmetic operators in algebra namespace

### PR DESCRIPTION
**Pull request by niermann999**
The Eigen Identity matrix in the transform class leads to invalid instructions in CUDA. Also fixes small issues in the matrix getters of the Vc and SMatrix plugins.

Also moves the arithmetic operators of the cmath plugin to the algebra namespace, since they were picked up in ACTS builds by Eigen types

### CODE REVIEW

File: math/eigen/include/algebra/math/impl/eigen_transform3.hpp [hunk 1]
```diff
@@ -81,6 +79,8 @@ struct transform3 {
   ALGEBRA_HOST_DEVICE
   transform3(const vector3 &t, const vector3 &x, const vector3 &y,
              const vector3 &z, bool get_inverse = true) {
+    _data.setIdentity();
```
Comment (beomki-yeo) [inline math/eigen/include/algebra/math/impl/eigen_transform3.hpp:82]: I think you also need to set `_data_inv` to identity matrix

File: storage/common/include/algebra/storage/matrix_getter.hpp [hunk 2]
```diff
@@ -241,7 +241,7 @@ ALGEBRA_HOST_DEVICE constexpr void set_block(
   if constexpr (ROWS == mROW &&
                 matrix_t::storage_rows() == input_matrix_t::storage_rows()) {
     if (row == 0u) {
-      for (std::size_t j = col; j < mCOL; ++j) {
+      for (std::size_t j = col; j < col + COLS; ++j) {
```
[the remark this answers is not in the document]
Comment (niermann999) [reply storage/common/include/algebra/storage/matrix_getter.hpp:244]: Not sure. The vector getter used ```SubCol``` previously, which worked in detray as long as we used it with ```row=0```. With any other row index, it starts to go out of bounds. So, I put in this code and made a mistake myself

File: math/eigen/include/algebra/math/impl/eigen_transform3.hpp (hunk 1 above)
Comment (niermann999) [reply math/eigen/include/algebra/math/impl/eigen_transform3.hpp:82]: Ah, right, I keep forgetting that ```get_inverse``` could be ```false```. At that point, accessing ```_data_inv``` should not be permitted actually, regardless if it is the identity matrix or garbage

File: storage/common/include/algebra/storage/matrix_getter.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (niermann999) [reply storage/common/include/algebra/storage/matrix_getter.hpp:244]: Ah, wait, this is the matrix getter for SoA types. This is new code

File: math/eigen/include/algebra/math/impl/eigen_transform3.hpp (hunk 1 above)
Comment (niermann999) [reply math/eigen/include/algebra/math/impl/eigen_transform3.hpp:82]: Done

**Review by beomki-yeo — APPROVED** [review]

**Review by krasznaa — APPROVED** [review]
Apart from the formatting issues, this looks fine to me.

Just as a comment: Instead of the `using namespace algebra` statements, I myself would've probably put the common test code into let's say an `algebra::test::` namespace.

But since it's testing code, I don't actually care a whole lot about the exact formalism...

**Pull request merged**